src/addrman_impl.h

```diff
@@ -178,8 +178,10 @@ class AddrManImpl
     //! @note Don't increment this. Increment `lowest_compatible` in `Serialize()` instead.
     static constexpr uint8_t INCOMPATIBILITY_BASE = 32;
 
-    //! last used nId
-    int nIdCount GUARDED_BY(cs){0};
+    //! last used nId.
+    //! This used to be an int, making it feasible for attackers to cause an overflow,
+    //! see https://bitcoincore.org/en/2024/07/31/disclose-addrman-int-overflow/
+    uint64_t nIdCount GUARDED_BY(cs){0};
```

Comment:
Amend the `pnId` parameter type in `AddrManImpl::Create` respectively?

Reply:
true. It also needs more changes (`mapInfo`). I wil put te PR to draft until I have addressed these.

Reply:
Should be addressed now.